---
title: "Extract ApoE from WGS data; all cohorts"
output:
  html_document:
    toc: true
    toc_depth: 2
    number_sections: false
---
 
### Get positions and REF/ALT from the annotation file for chromosome 19
```{r, eval=FALSE}
annot <- read.csv("../../Raw/Joint/WGS/NIA_JG_1898_samples_GRM_WGS_b37_JointAnalysis01_2017-12-08_19.recalibrated_variants.annotated.txt", sep="\t")
annot[which(annot$ID=="rs429358"),c("POS","REF","ALT")] # 45411941 T C
annot[which(annot$ID=="rs7412"),c("POS","REF","ALT")] # 45412079 C T
```

### Extract the genotypes for all samples from the original raw vcf file
https://en.wikipedia.org/wiki/Apolipoprotein_E

```{bash, eval=FALSE}
module load bcftools/1.8

InPath=../../Raw/Joint/WGS
OutPath=../../Reports/WGS_PrelimProcessing

bcftools query -f '[%POS %SAMPLE %GT\n]' -i '45411941' \
  $InPath/NIA_JG_1898_samples_GRM_WGS_b37_JointAnalysis01_2017-12-08_19.recalibrated_variants.vcf.gz > \
  $OutPath/ApoE1_rs429358.allCohorts.txt

bcftools query -f '[%POS %SAMPLE %GT\n]' -i '45412079' \
  $InPath/NIA_JG_1898_samples_GRM_WGS_b37_JointAnalysis01_2017-12-08_19.recalibrated_variants.vcf.gz > \
  $OutPath/ApoE2_rs7412.allCohorts.txt

```

### Translate to combined genotype
```{r, eval=FALSE}

ApoEp1 <- read.table("../../Reports/WGS_PrelimProcessing/ApoE_rs429358.allCohorts.txt", colClasses="character")
colnames(ApoEp1) <- c("Position", "WGS_SampleID", "rs429358")
ApoEp2 <- read.table("../../Reports/WGS_PrelimProcessing/ApoE_rs7412.allCohorts.txt", colClasses="character")
colnames(ApoEp2) <- c("Position", "WGS_SampleID", "rs7412")

ApoEgt <- merge(ApoEp1[2:3], ApoEp2[2:3], by="WGS_SampleID")
ApoEgt$ApoE_WGS <- NA
ApoEgt$ApoE_WGS[ApoEgt$rs429358=="0/0" & ApoEgt$rs7412=="1/1"] <- "22"
ApoEgt$ApoE_WGS[ApoEgt$rs429358=="0/0" & ApoEgt$rs7412=="1/0"] <- "23"
ApoEgt$ApoE_WGS[ApoEgt$rs429358=="0/1" & ApoEgt$rs7412=="1/0"] <- "24"
ApoEgt$ApoE_WGS[ApoEgt$rs429358=="0/0" & ApoEgt$rs7412=="0/0"] <- "33"
ApoEgt$ApoE_WGS[ApoEgt$rs429358=="0/1" & ApoEgt$rs7412=="0/0"] <- "34"
ApoEgt$ApoE_WGS[ApoEgt$rs429358=="1/1" & ApoEgt$rs7412=="0/0"] <- "44"
ApoEgt$ApoE_WGS[ApoEgt$rs429358=="0/0" & ApoEgt$rs7412=="0/1"] <- "23"
ApoEgt$ApoE_WGS[ApoEgt$rs429358=="0/1" & ApoEgt$rs7412=="0/1"] <- "24"

# Compare with reported genotype
#ApoEgt$ApoE_reported <- NA

mayo <- read.csv("../../Processed/Mayo/Metadata/Mayo_allMetadata.csv", colClasses = "character", na.strings = "")
mayo <- mayo[!is.na(mayo$WGS_SampleID), c("WGS_SampleID", "ApoE")]
colnames(mayo)[2] <- "ApoE_reported"
mayo <- unique(mayo)
mayo$Cohort <- "mayo"
length(which(is.na(mayo$ApoE_reported))) # 56 are missing ApoE genotype from clinical data

msbb <- read.csv("../../Processed/MSBB/Metadata/MSBB_allMetadata.csv", colClasses = "character", na.strings = "")
msbb$ApoE_reported <- paste0(msbb$Apo1, msbb$Apo2)
msbb$ApoE_reported[msbb$ApoE_reported=="NANA"] <- NA
msbb <- msbb[!is.na(msbb$WGS_SampleID), c("WGS_SampleID", "ApoE_reported")]
msbb <- unique(msbb)
msbb$Cohort <- "msbb"
length(which(is.na(msbb$ApoE_reported))) # 169 are missing ApoE genotype from clinical data

rosmap <- read.csv("../../Processed/ROSMAP/Metadata/ROSMAP_allMetadata.csv", colClasses = "character", na.strings = "")
rosmap <- rosmap[!is.na(rosmap$WGS_SampleID), c("WGS_SampleID", "apoe_genotype")]
colnames(rosmap)[2] <- "ApoE_reported"
rosmap <- unique(rosmap)
rosmap$Cohort <- "rosmap"
length(which(is.na(rosmap$ApoE_reported))) # 14 are missing ApoE genotype from clinical data

ApoEgt <- merge(ApoEgt, rbind(mayo, msbb, rosmap), by="WGS_SampleID", all=TRUE)

length(which(is.na(ApoEgt$ApoE_reported))) # 239 missing clinical report of ApoE; as above
length(di <- which(ApoEgt$ApoE_reported != ApoEgt$ApoE_WGS)) # 32 are different between reported and WGS
ApoEgt$Cohort[di] # all the different ones are from MSBB and ROSMAP
ApoEgt$ApoE_dif <- ApoEgt$ApoE_reported != ApoEgt$ApoE_WGS


write.csv(ApoEgt, file="../../Reports/WGS_PrelimProcessing/ApoE.all.csv", row.names = FALSE, na="")

```